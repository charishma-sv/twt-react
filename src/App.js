import React, { Component } from 'react';
import Application from './components/Application';
import UserProvider from './providers/UserProvider';
import './styles/scss/main.scss';
import CacheBuster from './CacheBuster';

class App extends Component {
  constructor() {
    super();
    this.state = {};
  }
  render() {
    return (
      <CacheBuster>
        {({ loading, isLatestVersion, refreshCacheAndReload }) => {
          if (loading) return null;
          if (!loading && !isLatestVersion) {
            refreshCacheAndReload();
          }
          return (
            <div>
              <UserProvider>
                <Application />
              </UserProvider>
            </div>
          );
        }}
      </CacheBuster>
    );
  }
}

export default App;
