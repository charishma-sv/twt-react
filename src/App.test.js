import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';

describe('snapshot <App />', () => {
  it('should render snapshot', () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
