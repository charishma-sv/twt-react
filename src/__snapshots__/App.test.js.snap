// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot <App /> should render snapshot 1`] = `null`;
